File: app/src/main/java/ir/mymessage/view/base/BasePresenter.java
package ir.mymessage.view.base;

import android.content.Context;

import ir.mymessage.service.ApiClient;
import ir.mymessage.service.ApiInterface;
import ir.mymessage.utils.MySharedPrefrences;

public abstract class BasePresenter {

    //protected RestClient restClient = new RestClient();
    //protected DBClient dbClient;
    protected Context context;
    protected ApiInterface apiService = ApiClient.getRetrofit().create(ApiInterface.class);

}

File: app/src/main/java/ir/mymessage/view/base/BasePresenter.kt
package ir.mymessage.view.base

import android.content.Context

import ir.mymessage.service.ApiClient
import ir.mymessage.service.ApiInterface
import ir.mymessage.utils.MySharedPrefrences

abstract class BasePresenter {
    protected var context: Context? = null
    protected var apiService = ApiClient.getRetrofit()!!.create(ApiInterface::class.java)

}
